astropy/units/tests/test_logarithmic.py

```diff
@@ -832,30 +832,7 @@ def test_dot(self):
 
     @pytest.mark.parametrize('method', ('prod', 'cumprod'))
     def test_never_ok(self, method):
-        with pytest.raises(ValueError):
+        with pytest.raises(TypeError):
             getattr(self.mJy, method)()
-
-        with pytest.raises(ValueError):
+        with pytest.raises(TypeError):
             getattr(self.m1, method)()
-
-
-class TestLogQuantityUfuncs:
```

Comment:
The reason I removed it is that, really, ufuncs are not done well altogether. It makes little sense for `np.power` to work when `np.add` does not. For future work that needs to be done, see #6897 and in particular https://github.com/astropy/astropy/issues/6897#issuecomment-453217385